flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/resource/ResourceManager.java

```diff
@@ -203,54 +193,75 @@ public void close() throws IOException {
         }
     }
 
-    private void checkJarResources(List<ResourceUri> resourceUris) throws IOException {
-        // only support register jar resource
-        if (resourceUris.stream()
-                .anyMatch(resourceUri -> !ResourceType.JAR.equals(resourceUri.getResourceType()))) {
-            throw new ValidationException(
-                    String.format(
-                            "Only support to register jar resource, resource info:\n %s.",
-                            resourceUris.stream()
-                                    .map(ResourceUri::getUri)
-                                    .collect(Collectors.joining(",\n"))));
-        }
+    /** Check whether the {@link Path} exists. */
+    public boolean exists(Path filePath) throws IOException {
+        return filePath.getFileSystem().exists(filePath);
+    }
 
-        for (ResourceUri resourceUri : resourceUris) {
-            checkJarPath(new Path(resourceUri.getUri()));
+    /**
+     * Synchronize a file resource identified by the given {@link ResourceUri} with a local copy
```

Comment:
> The path passed to resourceGenerator will be a local path received from the given {@link ResourceUri}.

I think this description is not so accurate because if the given resource URI is `hdfs://foo/bar/plan.json`, then the local path passes to the resource generator should be `file://root-dir-forresource-manager/plan-uuid.json`